lib/phoenix_live_view/utils.ex

```diff
@@ -284,21 +284,37 @@ defmodule Phoenix.LiveView.Utils do
     """
   end
 
+  @doc """
+  Returns a map of infos about the lifecycle stage for the given `view`.
+  """
+  def lifecycle(%Socket{} = socket, view, stage, arity) do
+    info = %{
+      callbacks?: Lifecycle.callbacks?(socket, stage),
+      exported?: function_exported?(view, stage, arity),
+      view: view
+    }
+
+    Map.put(info, :any?, info.callbacks? or info.exported?)
```

Comment:
Better to create a map at once instead of creating and then putting:

```suggestion
    callbacks? = Lifecycle.callbacks?(socket, stage)
    exported? = function_exported?(view, stage, arity)

    %{
      any?: callbacks? or exported?,
      callbacks?: callbacks?,
      exported?: exported?,
      view: view
    }
```

:)

Btw, do you think this function should exist in the lifecycle module?

Reply:
> do you think this function should exist in the lifecycle module?

That seems like a good place for it. `Lifecycle.info/1` or similar?